pkg/util/oc/oc_struct.go

```diff
@@ -71,20 +72,51 @@ func (o OC) ApplyTemplateFile(t test.TestHelper, ns string, tmplFile string, inp
 	})
 }
 
+// This function can be called by both ApplyString and ApplyFile. Accept as argument: multiple yaml string or yaml file to apply
+func (o OC) applyWithRetry(t test.TestHelper, ns string, inputKind string, arg string) {
+	t.T().Helper()
+	maxAttempts := 5
+	var attemptT *test.RetryTestHelper
+	warning := false
+	for i := 0; i < maxAttempts; i++ {
+		attemptT = retry.Attempt(t, func(t test.TestHelper) {
+			t.T().Helper()
+			o.withKubeconfig(t, func() {
+				t.T().Helper()
+				if inputKind == "string" {
+					shell.ExecuteWithInput(t, fmt.Sprintf("oc %s apply -f -", nsFlag(ns)), arg)
+				} else if inputKind == "file" {
+					o.Invokef(t, "oc %s apply -f %s", nsFlag(ns), arg)
+				} else {
+					t.Fatalf("Invalid inputKind: %s", inputKind)
+				}
+			})
+		})
+		if !attemptT.Failed() {
+			if warning {
+				t.Logf("WARNING: Apply attempt %d of %d succeeded after previous failures.", i+1, maxAttempts)
```

Comment:
Do we need this log? If the previous attempt failed, I will see an error in logs, so if the next iteration will succeed, it will be clear that the succeeded after X failures.

Reply:
@luksa suggests only logging as a warning if fails one time in the retry loop, and if fails all the attempts log as failures.